cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +796,21 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+int32_t gdv_fn_instr_utf8(int64_t context, const char* string, int32_t string_len,
+                         const char* substring, int32_t substring_len) {
+  if (substring_len == 0) {
+    return 1;
+  }
+
+  for(int i = 0; i < string_len; i++) {
+    if (string[i] == substring[0] && gandiva::string_matches(string + 1, substring + 1, i, substring_len - 1)) {
```

Comment:
Please use memcmp instead - https://www.cplusplus.com/reference/cstring/memcmp/

You are essentially comparing the next substring_len - 1 bytes of both the pointers

Reply:
It is using memcmp now